### PR TITLE
Implement `ReflectionClass#getInterfaceClassNames()`

### PR DESCRIPTION
**Pull request by staabm**
same motivation as in https://github.com/Roave/BetterReflection/pull/1359, but this time for interfaces :-)

refs https://github.com/rectorphp/rector/issues/8093#issuecomment-1657057947

---

if we agree on this one, I should do another PR for `traits` as a followup

### CODE REVIEW

File: test/unit/Reflection/ReflectionClassTest.php [hunk 1]
```diff
@@ -425,6 +426,93 @@ public function testGetParentClassWithMissingParent(): void
...
+    {
+        $classInfo = (new DefaultReflector(new SingleFileSourceLocator(
+            __DIR__ . '/../Fixture/ClassWithMissingInterface.php',
+            $this->astLocator,
+        )))->reflectClass(ClassWithMissingInterface::class);
+
+        $this->expectException(IdentifierNotFound::class);
+
+        self::assertNotNull($classInfo->getInterfaces());
+    }
```
Comment (staabm) [inline test/unit/Reflection/ReflectionClassTest.php:451]: these tests show, that the pre-PR existing API always throws and therefore does not allow to get the interface names, when the class-symbols are not autoloadable.

File: test/unit/Reflection/ReflectionClassTest.php [hunk 2]
```diff
@@ -425,6 +426,93 @@ public function testGetParentClassWithMissingParent(): void
...
+
+    /** @return list<array{string, class-string, list<class-string>}> */
+    public static function getInterfaceClassNamesDataProvider(): array
+    {
+        return [
+            [
+                __DIR__ . '/../Fixture/ClassWithMissingInterface.php',
+                ClassWithMissingInterface::class,
+                ['Roave\BetterReflectionTest\Fixture\InterfaceThatDoesNotExist'],
+            ],
```
Comment (staabm) [inline test/unit/Reflection/ReflectionClassTest.php:476]: this first test-case shows that we are now able to extract the interface names even if not autoloadable.

the remaining cases add coverage for a few more related cases like autoloadable classes, traits,..

**Comment by Ocramius** [comment]
This is a new feature: should go to `6.12.x` :-)

**Review by Ocramius — APPROVED** [review]
LGTM: feel free to send more of these, BTW.

**Comment by staabm** [comment]
oh sorry. did not relealize 6.12.x is already open

**Comment by Ocramius** [comment]
Yeah, release automation does create a new branch each time :)

**Comment by Ocramius** [comment]
Thanks @staabm!

**Comment by staabm** [comment]
thank you. I am already working on the "trait-variant" of it

**Pull request merged**